eslint/babel-eslint-plugin/src/rules/no-unused-expressions.js

```diff
@@ -31,7 +32,8 @@ function isInDoStatement(node) {
   if (
     node.type === "IfStatement" &&
     node.parent &&
-    node.parent.type === "IfStatement"
+    node.parent.type === "IfStatement" &&
+    node.parent.alternate === node
```

Comment:
Is that fixing a bug? I suggest we do the refactor only and isolate bug fix to another PR.

Reply:
Yes, this is guarding against `if (foo) if (baz) console.log('hi friend');`.

I'm not against removing this change, but a lot of test cases actually throw parser errors with the updated parser (this plugin is reeeaaaaaallly out of date 😬 ), so it seemed easier to update it in one fell swoop.

Reply:
But I can definitely remove the bug fixes/refactors from this PR if you prefer!

Reply:
Also, want to note that this also includes the changes from https://github.com/babel/babel/pull/10975, so the diff isn't actually this big 😄